src/UNL/UCBCN/Manager/CreateCalendar.php

```diff
@@ -31,6 +31,34 @@ public function __construct($options = array())
         }
     }
 
+    public function handlePost(array $get, array $post, array $files)
+    {
+        if (array_key_exists('calendar_shortname', $this->options)) {
+            # we are editing an existing calendar
+            $this->calendar = Calendar::getByShortname($this->options['calendar_shortname']);
+
+            if ($this->calendar === FALSE) {
+                throw new \Exception("That calendar could not be found.", 404);
+            }
+
+            $this->updateCalendar($post);
+            $this->flashNotice('success', 'Calendar Updated', 'Your calendar "' . $this->calendar->name . '" has been updated.');
+        } else {
+            # we are creating a new calendar
+            try {
+                $this->calendar = $this->createCalendar($post);    
+            } catch (ValidationException $e) {
+                $this->flashNotice('failure', 'Sorry! We couldn\'t create your calendar', $e->getMessage());
+                return Controller::$url . 'calendar/new/';
```

Comment:
Am I correct that this would result in a 200 HTTP status?  A failure due to bad user data should probably not result in a success status.  Also, if you don't redirect here, you could still have access to the POST data which you could then use to auto-fill the form.